packages/x-scheduler-headless/src/utils/recurring-events/getRecurringEventOccurrencesForVisibleDays.ts

```diff
@@ -68,20 +69,27 @@ class RecurringEventExpander {
   constructor(
     private readonly adapter: Adapter,
     private readonly event: SchedulerProcessedEvent,
+    private readonly uiTimezone: TemporalTimezone,
     start: TemporalSupportedObject,
     end: TemporalSupportedObject,
   ) {
-    this.rule = event.rrule!;
-    this.seriesStart = adapter.startOfDay(event.start.value);
+    this.rule = event.modelInBuiltInFormat!.rrule! as RecurringEventRecurrenceRule;
+    this.seriesStart = adapter.startOfDay(event.modelInBuiltInFormat!.start);
     this.interval = Math.max(1, this.rule.interval ?? 1);
 
+    const dataTz = adapter.getTimezone(event.modelInBuiltInFormat!.start);
```

Comment:
Nitpick, you use `event.modelInBuiltInFormat!` several time, I would save it in a class property to avoid using `!` for every access

Reply:
Can you also create an issue to improve the Typing and have a non nullable `modelInBuiltInFormat` (and a distinct interface for the event creation that doesn't have it, it's the only reason why it's nullable AFAIK).